test/e2e/csv_e2e_test.go

```diff
@@ -4175,58 +4175,61 @@ var _ = Describe("ClusterServiceVersion", func() {
 })
 
 var _ = Describe("Disabling copied CSVs", func() {
-	// Define namespace, operatorGroup, and csv upfront
-	ns := &corev1.Namespace{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: genName("csv-toggle-test-"),
-		},
-	}
+	var (
+		ns corev1.Namespace
+		csv operatorsv1alpha1.ClusterServiceVersion
 
-	operatorGroup := operatorsv1.OperatorGroup{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      genName("csv-toggle-test-"),
-			Namespace: ns.GetName(),
-		},
-	}
+	)
 
-	csv := operatorsv1alpha1.ClusterServiceVersion{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      genName("csv-toggle-test-"),
-			Namespace: ns.GetName(),
-		},
-		Spec: operatorsv1alpha1.ClusterServiceVersionSpec{
-			InstallStrategy: newNginxInstallStrategy(genName("csv-toggle-test-"), nil, nil),
-			InstallModes: []operatorsv1alpha1.InstallMode{
-				{
-					Type:      operatorsv1alpha1.InstallModeTypeAllNamespaces,
-					Supported: true,
-				},
+	BeforeEach(func() {
+		nsname := genName("csv-toggle-test-")
+		og := operatorsv1.OperatorGroup{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      fmt.Sprintf("%s-operatorgroup", nsname),
+				Namespace: nsname,
 			},
-		},
-	}
+		}
+		ns = SetupGeneratedTestNamespaceWithOperatorGroup(nsname, og)
 
-	When("an operator is installed in AllNamespace mode", func() {
-		BeforeEach(func() {
-			Eventually(func() error {
-				if err := ctx.Ctx().Client().Create(context.TODO(), ns); err != nil && !k8serrors.IsAlreadyExists(err) {
-					ctx.Ctx().Logf("Unable to create ns: %v", err)
-					return err
-				}
+		Eventually(func() error {
+			csv = operatorsv1alpha1.ClusterServiceVersion{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      genName("csv-toggle-test-"),
+					Namespace: nsname,
+				},
+				Spec: operatorsv1alpha1.ClusterServiceVersionSpec{
+					InstallStrategy: newNginxInstallStrategy(genName("csv-toggle-test-"), nil, nil),
+					InstallModes: []operatorsv1alpha1.InstallMode{
+						{
+							Type:      operatorsv1alpha1.InstallModeTypeAllNamespaces,
+							Supported: true,
+						},
+					},
+				},
+			}
 
-				if err := ctx.Ctx().Client().Create(context.TODO(), &operatorGroup); err != nil && !k8serrors.IsAlreadyExists(err) {
-					ctx.Ctx().Logf("Unable to create og: %v", err)
-					return err
-				}
+			if err := ctx.Ctx().Client().Create(context.TODO(), &csv); err != nil && !k8serrors.IsAlreadyExists(err) {
+				ctx.Ctx().Logf("Unable to create csv: %v", err)
+				return err
+			}
 
-				if err := ctx.Ctx().Client().Create(context.TODO(), &csv); err != nil && !k8serrors.IsAlreadyExists(err) {
-					ctx.Ctx().Logf("Unable to create csv: %v", err)
-					return err
-				}
+			return nil
+		}).Should(Succeed())
 
-				return nil
-			}).Should(Succeed())
-		})
+	})
+	AfterEach(func() {
+		Eventually(func() error {
+			err := ctx.Ctx().Client().Delete(context.TODO(), &csv)
```

Comment:
(this is applicable to all context.TODO references in this PRs' changes)

nit: can we use context.Background() here instead of context.TODO() to be consistent with the rest of the csv_e2e_test.go testing file?

Reply:
I'll make changes.  Thanks!